### DISCUSSION TITLE
Use enumerations for Error kinds

### DISCUSSION THREAD

**Discussion by aesteve** [scala/scala3#17898] [Feature Requests]
For a newcomer implementing Error Messages, choosing the right kind of error is a bit difficult. What is "kind" ? What does it mean ? Is it documented somewhere ? Which one should I pick ?

A very simple first step could be to gather all the existing kinds into an enumeration, this is simple, and allows to gather every kind of error in a single place.

This would allow newcomers to, at least, quickly get a glimpse of the different kind of errors already used (`Syntax`, `Reference`, etc.). 

After that, maybe documenting the enum would be a very good idea.

**Comment by propensive** [comment]
Looks good. Could we call them "categories" of errors, rather than "kinds"?

**Comment by som-snytt** [comment]
`-Wconf` nomenclature is "category".
